secure/json-rpc/src/lib.rs

```diff
@@ -0,0 +1,595 @@
+                if let Ok(failure_response) =
+                    serde_json::from_str::<JSONRpcFailureResponse>(response)
+                {
+                    Err(Error::InternalRPCError(format!("{:?}", failure_response)))
+                } else {
+                    let _submit_response =
+                        serde_json::from_str::<SubmitTransactionResponse>(response)?;
+                    Ok(())
+                }
+            }
+            _ => Err(Error::RPCFailure(response.into_string()?)),
+        }
+    }
+
+    /// Returns the associated AccountState for a specific account at a given version height
+    /// using the JSON RCP API.
+    pub fn get_account_state(
+        &self,
+        account: AccountAddress,
+        version: u64,
+    ) -> Result<AccountState, Error> {
+        let response = self.get_account_state_with_proof(account, version, version)?;
+        match response.status() {
+            200 => {
+                let response = &response.into_string()?;
+                if let Ok(failure_response) =
+                    serde_json::from_str::<JSONRpcFailureResponse>(response)
+                {
+                    Err(Error::InternalRPCError(format!("{:?}", failure_response)))
+                } else if let Some(blob_bytes) =
+                    serde_json::from_str::<AccountStateWithProofResponse>(response)?
+                        .result
+                        .blob
+                {
+                    let account_state_blob = AccountStateBlob::from(lcs::from_bytes::<Vec<u8>>(
+                        &*blob_bytes.into_bytes()?,
+                    )?);
+                    if let Ok(account_state) = AccountState::try_from(&account_state_blob) {
+                        Ok(account_state)
+                    } else {
+                        Err(Error::SerializationError(format!(
+                            "Unable to convert account_state_blob to AccountState: {:?}",
+                            account_state_blob
+                        )))
+                    }
+                } else {
+                    Err(Error::MissingData("AccountState".into()))
+                }
+            }
+            _ => Err(Error::RPCFailure(response.into_string()?)),
+        }
+    }
+
+    // Executes the specified request method using the given parameters by contacting the JSON RPC
+    // server.
+    fn execute_request(&self, method: String, params: Vec<Value>) -> Response {
+        ureq::post(&self.host)
+            .timeout_connect(REQUEST_TIMEOUT)
+            .send_json(
+                json!({"jsonrpc": JSON_RPC_VERSION, "method": method, "params": params, "id": 0}),
+            )
+    }
+
+    // Sends a submit() request to the JSON RPC server using the given transaction.
+    fn submit_transaction(&self, signed_transaction: SignedTransaction) -> Result<Response, Error> {
+        let method = "submit".to_string();
+        let params = vec![Value::String(hex::encode(lcs::to_bytes(
```

Comment:
I wonder why they went with hex instead of base64....

Reply:
Yeah.. I saw the discussion you started yesterday!